src/plugins/intel_cpu/src/nodes/scatter_update.h

```diff
@@ -49,22 +99,32 @@ class ScatterUpdate : public Node {
     void scatterElementsUpdate(const MemoryPtr& mem_data, const MemoryPtr& mem_indices, const MemoryPtr& mem_updates, int axis, const func_t& kernel_func);
     template <typename DataType>
     void scatterElementsUpdate(const MemoryPtr& mem_data, const MemoryPtr& mem_indices, const MemoryPtr& mem_updates,
-                                int axis, const ReduceMean& kernel_func);
-    template <typename DataType>
-    inline void scatterElementsUpdate_dispatch(const MemoryPtr& dstMemPtr, const MemoryPtr& indicesMemPtr, const MemoryPtr& updateMemPtr, int axis);
+                                int axis, const scatter_elements_update::ReduceMean& kernel_func);
 
     struct ScatterElementsUpdateContext {
         ScatterUpdate* node;
         MemoryPtr dstMemPtr;
         MemoryPtr indicesMemPtr;
         MemoryPtr updateMemPtr;
         int axis;
+        scatter_elements_update::ReduceBase* reduce;
     };
 
+    // tier 1 dispatcher with DataType
+    template <typename DataType>
+    inline void scatterElementsUpdate_dispatch(ScatterElementsUpdateContext& ctx);
     template<typename DataType>
     struct ScatterElementsUpdateDispatcher {
         void operator()(ScatterElementsUpdateContext& ctx) {
-            ctx.node->scatterElementsUpdate_dispatch<DataType>(ctx.dstMemPtr, ctx.indicesMemPtr, ctx.updateMemPtr, ctx.axis);
+            ctx.node->scatterElementsUpdate_dispatch<DataType>(ctx);
+        }
+    };
+    // tier 2 dispatcher with Reduce which follows up DataType.
+    template<typename PT>
+    struct ScatterElementsUpdateReduceDispatcher {
+        void operator()(ScatterElementsUpdateContext& ctx) {
+            ctx.node->scatterElementsUpdate<typename PT::first_type>(ctx.dstMemPtr, ctx.indicesMemPtr, ctx.updateMemPtr, ctx.axis,
+                                                                    static_cast<const typename PT::second_type&>(*ctx.reduce));
```

Comment:
Isn't it simpler and more easy to compiler just to have:

```cpp
    template<typename KernelType>
    struct ScatterElementsUpdateReduceDispatcher {
        ctx.node->scatterElementsUpdate<typename PT::first_type>(ctx.dstMemPtr, ctx.indicesMemPtr, ctx.updateMemPtr, ctx.axis,
                                                                    PT::second_type{});
    };
```
Why do we even need to explicitly pass the address of the kernel object?
Each kernel type has a default constructor (which does nothing) and a method with static address, the compiler will simply inline the kernel code in the end. Also we don't need to have static instances of the kernel classes at all as they don't have any state.
Also the base class will be redundant, as well as the kernel pointer in the context.
Moreover, with such a model you may make the predicate class to be parametrized with the template parameter rather than its `operator()`.

```cpp
    template<typename KernelType>
    struct ScatterElementsUpdateReduceDispatcher {
        using kernel_t = PT::second_type;
        using data_t = PT::first_type;
        ctx.node->scatterElementsUpdate<data_t>(ctx.dstMemPtr, ctx.indicesMemPtr, ctx.updateMemPtr, ctx.axis,
                                                                    kernel_t<data_t>{});
    };
```

Reply:
Great point! Refactored with some difference though -
1. We still need operator(), as we need to pass ctx, and OV_CASE needs it too. ref: https://github.com/openvinotoolkit/openvino/blob/ca5a69eea01bee4fc0011bd034027f5a545bbace/src/common/conditional_compilation/include/openvino/cc/selective_build.h#L96
2. PT::second_type should be a type, cannot be used as a template like -
`kernel_t<data_t>{}`

Reply:
> We still need operator(), as we need to pass ctx, and OV_CASE needs it too. ref:

I didn't mention that we need to remover `operator()` simply skipped it in the code snippet for simplicity

> PT::second_type should be a type, cannot be used as a template like - `kernel_t<data_t>`

Yep, my inattention, but anyway the main idea is that `PT::second_type` may be set as `kernel_t<data_t>` with a few tricks in ScatterElementsUpdateReduceDispatcher.

> The dispatchers as well as context are moved outside of class, at the cost of declaring the inner implementation of scatterElementsUpdate to be public.

Don't see any issues with that as nodes are not a part of the plugin interface.